core/src/main/boilerplate/tapir/typelevel/ParamsAsArgs.scala.template

```diff
@@ -0,0 +1,47 @@
+package tapir.typelevel
+
+import scala.annotation.implicitNotFound
+
+/**
+  * View parameters (single type or a tuple) as arguments of a function of the appropriate arity.
+  */
+trait ParamsAsArgs[I] {
+  // FN[R] = (I as args) => R
+  type FN[_]
+
+  def toFn[O](f: I => O): FN[O]
+  def paramAt(params: I, i: Int): Any
+  def applyFn[R](f: FN[R], args: I): R
+}
+
+object ParamsAsArgs extends LowPriorityParamsAsArgs1 {
+
+  [#implicit def tuple1ToFn[[#A1#]]: Aux[Tuple1[[#A1#]], ([#A1#]) => ?] = new ParamsAsArgs[Tuple1[[#A1#]]] {
```

Comment:
I've added some tests: https://github.com/softwaremill/tapir/commit/4724e46a45c86cfaffe5692ae0abcb1c634a38d6 which I think might fail due to double definition of `ParamsAsArgs` for the single-parameter case

Reply:
This might work for the same reason as before. 

And there's no way of generating instances from tuple2 upwards?

Reply:
Luckily there is. I thought that there isn't because I got confused by seeing an unmerged PR which adds such functionality.